### PR TITLE
Don't send plaintext reset passwords via email

### PR DESCRIPTION
**Pull request by paglias**
WIP

### CODE REVIEW

File: website/common/locales/en/front.json [hunk 1]
```diff
@@ -253,10 +253,11 @@
...
     "passwordResetEmailSubject": "Password Reset for Habitica",
-    "passwordResetEmailText": "Password for <%= username %> has been reset to <%= newPassword %> . Important! Both username and password are case-sensitive -- you must enter both exactly as shown here. We recommend copying and pasting both instead of typing them. Log in at <%= baseUrl %>. After you have logged in, head to <%= baseUrl %>/#/options/settings/settings and change your password.",
-    "passwordResetEmailHtml": "Password for <strong><%= username %></strong> has been reset to <strong><%= newPassword %></strong><br /><br />Important! Both username and password are case-sensitive -- you must enter both exactly as shown here. We recommend copying and pasting both instead of typing them.<br /><br />Log in at <%= baseUrl %>. After you have logged in, head to <%= baseUrl %>/#/options/settings/settings and change your password.",
+    "passwordResetEmailText": "If you requested a password reset for <%= username %> on Habitica head to <%= passwordResetLink %> to set a new one. The link will expire after 24 hours. If you haven't request a password reset please ignore this email.",
```
Comment (paglias) [inline website/common/locales/en/front.json]: @Alys can you review these strings and tell me if they're okay or if I should change them? Thanks!

Comment (Alys) [reply website/common/locales/en/front.json]: It's good! Just "request" in the final sentence should be "requested" and I'd put a comma after "Habitica".

Comment (khipkin) [reply website/common/locales/en/front.json]: I would also add a comma after "reset" in the last sentence.

Comment (paglias) [reply website/common/locales/en/front.json]: done :)

File: website/server/controllers/top-level/auth.js [hunk 2]
```diff
@@ -2,45 +2,94 @@ import locals from '../../middlewares/locals';
...
 
 // Internal authentication routes
 
-// Set a new password after having requested a password reset
+// TODO
+// - convert to bcrypt
+// - tests
+
+// returns the user if a valid password reset code is supplied, otherwise false
+async function validatePasswordResetCodeAndFindUser (code) {
```
Comment (TheHollidayInn) [inline website/server/controllers/top-level/auth.js]: You should move these functions out to a library (maybe the password utils) so they can be tested.

Comment (paglias) [reply website/server/controllers/top-level/auth.js]: Yes, good idea, will do it tomorrow

**Comment by paglias** [comment]
TODO: invalidate after use

**Comment by paglias** [comment]
Almost done, missing only correct rendering of messages

**Pull request merged**